lib/sycamore/sycamore/query/execution/physical_operator.py

```diff
@@ -57,12 +59,12 @@ class MathOperator(PhysicalOperator):
     Perform basic math on 2 input variables.
     """
 
-    def __init__(self, logical_node: LogicalOperator, query_id: str, inputs: List[Any]) -> None:
+    def __init__(self, logical_node: Math, query_id: str, inputs: List[Any]) -> None:
         super().__init__(logical_node, query_id, inputs)
 
     def execute(self) -> Any:
-        assert self.logical_node.data is not None
-        operator = self.logical_node.data.get("type")
+        assert isinstance(self.logical_node, Math)
```

Comment:
asserted by the construction already?

Reply:
Sadly this isn't good enough in case the field is reassigned.